include/xtensor/xutils.hpp

```diff
@@ -891,6 +892,85 @@ namespace xt
      */
     template <class T>
     using squared_norm_type_t = typename squared_norm_type<T>::type;
+
+    namespace alloc_tracking
+    {
+        static bool enabled = false;
+
+        static void enable()
+        {
+            enabled = true;
+        }
+
+        static void disable()
+        {
+            enabled = false;
+        }
+    }
+
+    template <typename T>
+    struct tracking_allocator
```

Comment:
You may want to track allocation while using aligned allocations, so the `tracking allocator` should be a 'decorator' of another allocator (this latter being passed as a template parameter).

You can use the private inheritance + using declarations idiom to avoid typing the forwarding of most of the methods.

Reply:
Hmm, in my opinion you'll only want the tracking allocator in Debug mode and therefore you will also not really care about the speed + alignment of allocs. But the idea with inheritance seems simple enough to give it a try.